email/src/main/java/io/micronaut/email/TransactionalEmail.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.email;
+
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Representation of a transactional email.
+ * @author Sergio del Amo
+ * @since 1.0.0
+ */
+@Introspected
+public class TransactionalEmail {
+
+    @NonNull
+    @NotNull
+    @Valid
+    private final Sender sender;
+
+    @NonNull
+    @NotNull
+    @Valid
+    private final Recipient recipient;
```

Comment:
It feels strange to use here the names `sender` and `recipient` and then use `from` and `to`.